unified-codes/libs/data/src/lib/types/Entity.ts

```diff
@@ -4,20 +4,23 @@ export interface IEntity {
   code: string;
   description: string;
   type: string;
-  properties?: Property[];
+  has_property?: Property[];
+  has_child?: IEntity[];
```

Comment:
I think dgraph syntax of relations might be causing confusion! think this should be `properties` and `children` (although not too sure about having a recursive interface, but think it's OK!). If you need queries, can add them to the `Entity` class, e.g.

```
hasChild(child: IEntity) { 
  return this.children.map((child: IEntity) => child.code).includes(child.code); 
}
```

or something like that (or could also continue with existing pattern with something like `matchesChild`)?

Reply:
Yeah, I do agree children and properties sound nicer. I think the GraphQL API (the resolver anyway) needs some tweaking before we can use those names though - I was getting no results when using `children` and `properties` 😞